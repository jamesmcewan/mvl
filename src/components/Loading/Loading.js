import React from 'react';
import styled, { keyframes } from 'styled-components';

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const LoadingContainer = styled.div`
  display: flex;
  margin: 52px auto;
`;

const Gear = styled.div`
  animation: ${rotate360} 2s linear infinite;
  font-size:100px;
  width:100px;
`

const Loading = () => (
  <LoadingContainer>
    <Gear>âš™</Gear>
  </LoadingContainer>
);

export default Loading;
