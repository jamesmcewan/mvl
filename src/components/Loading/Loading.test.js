import React from 'react';
import { mount } from 'enzyme';
import Loading from './Loading';
import 'jest-styled-components';

describe('Loading', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<Loading />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });
});
