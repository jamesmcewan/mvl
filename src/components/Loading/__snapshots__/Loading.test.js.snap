// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render the correct JSX 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 52px auto;
}

.c1 {
  -webkit-animation: iVXCSc 2s linear infinite;
  animation: iVXCSc 2s linear infinite;
  font-size: 100px;
  width: 100px;
}

<Loading>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          âš™
        </div>
      </styled.div>
    </div>
  </styled.div>
</Loading>
`;
